File: python_programming_basics/while_loops/e_04_walking.py
GOAL = 10_000
total_steps = 0

while total_steps < GOAL:
    steps = input()

    if steps == "Going home":
        total_steps += int(input())
        break

    total_steps += int(steps)

if total_steps >= GOAL:
    print(f"Goal reached! Good job!\n{total_steps - GOAL} steps over the goal!")

else:
    print(f"{GOAL - total_steps} more steps to reach goal.")

File: JavaScript Basics/05. While Loop/e-04-Walking.js
function walking(input) {
    let goal = 10000;
    let totalSteps = 0;
    let index = 0;

    while (index < input.length) {
        let command = input[index];
        index++;

        if (command === "Going home") {

            let homeSteps = Number(input[index]);
            totalSteps += homeSteps;
            break;
        
        } else {
            let steps = Number(command);
            totalSteps += steps;
        }

        if (totalSteps >= goal) {
            console.log("Goal reached! Good job!");
            console.log(`${totalSteps - goal} steps over the goal!`);
            return;
        }
    }

    if (totalSteps < goal) {
        console.log(`${goal - totalSteps} more steps to reach goal.`);
    } else {
        console.log("Goal reached! Good job!");
        console.log(`${totalSteps - goal} steps over the goal!`);
    }
}